generate-readme.ts

```diff
@@ -0,0 +1,73 @@
+import fs from 'node:fs/promises'
+import path from 'node:path'
+
+const submodules = [
+  path.join('@seamapi', 'webhook'),
+  path.join('@seamapi', 'http'),
+]
+
+const sections = await Promise.all(submodules.map(readUsageSection))
+await writeReadmeUsage(sections.join('\n'))
+
+async function readUsageSection(modulePath: string): Promise<string> {
+  const data = await fs.readFile(
+    path.join('node_modules', modulePath, 'README.md'),
+    {
+      encoding: 'utf-8',
+    },
+  )
+
+  const regex = /#* Usage\s*([\s\S]*?(?=\n## \w))/
+  const matches = regex.exec(data)
+  if (matches == null || matches.length !== 2) {
+    throw new Error('Missing [## Usage] section')
+  }
+
+  const usage = matches[1]?.replace(/^\s+|\s+$/g, '') // trim any trailing whitespace
+
+  if (usage == null) {
+    throw new Error('Invalid [## Usage] format')
+  }
+
+  return usage
+}
+
+async function writeReadmeUsage(content: string): Promise<void> {
+  const projectReadme = await fs.readFile('./README.md', {
+    encoding: 'utf-8',
+  })
+
+  const usageRegex = /### Usage\s*([\s\S]*?(?=\n## Development))/
+
+  const matches = usageRegex.exec(projectReadme)
+
+  if (matches == null || matches.length !== 2 || matches[1] == null) {
+    throw new Error('Invalid README.md format')
+  }
+
+  const updatedContent = content
+    .replaceAll('@seamapi/webhook', 'seam')
+    .replaceAll('@seamapi/http', 'seam')
+    .replaceAll('SeamHttp', 'Seam')
+    .replaceAll('seam/connect', 'seam')
+
+  const currentUsageSection = matches[1]
+
+  if (
+    currentUsageSection
+      .replaceAll(/\s/g, '')
+      .includes(updatedContent.replaceAll(/\s/g, '')) // Remove all whitespace to ignore formatting changes
+  ) {
+    return
+  }
+
+  const injected = `### Usage
+
+${currentUsageSection}
```

Comment:
@razor-x , I just realized this implementation will just append on every run, and include old content.

Can we just replace the `## Usage` in this repo instead?

![CleanShot 2024-03-28 at 10 17 21@2x](https://github.com/seamapi/javascript-next/assets/11449462/52311719-a350-46c4-9f8d-aefa337bcff3)

```suggestion
```

Reply:
Yes, good call. We should update the http README with this example then. We can do that after this merges to test the end to end flow 🙂

Reply:
I realized this repo has `### Usage` but should have `## Usage` too

Reply:
I updated javascript-http README to have these examples now